subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_rai.c

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/net/socket.h>
+#include <zephyr/net/lwm2m.h>
+#include <modem/lte_lc.h>
+#include <nrf_modem_at.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(lwm2m_rai, CONFIG_LWM2M_CLIENT_UTILS_LOG_LEVEL);
+static void lte_rrc_event_handler(const struct lte_lc_evt *const evt);
+
+static bool rrc_connected;
+
+static int set_rel14feat(void)
+{
+	int ret;
+
+	ret = nrf_modem_at_printf("AT%%REL14FEAT=0,1,0,0,0");
+	if (ret) {
+		LOG_ERR("AT command failed, error code: %d", ret);
+		return -EFAULT;
+	}
+
+	return 0;
+}
+
+static int read_rai(void)
+{
+	uint16_t rai;
+	int ret;
+
+	ret = nrf_modem_at_scanf("AT%RAI?", "%%RAI: %hu", &rai);
+	if (ret != 1) {
+		LOG_ERR("Failed to read RAI, error code: %d", ret);
+		return ret;
+	}
+	LOG_DBG("RAI: %d", rai);
+	return 0;
+}
+
+static int rai_req(bool enable)
+{
+	int ret;
+
+	ret = nrf_modem_at_printf("AT%%RAI=%d", enable);
+	if (ret) {
+		LOG_ERR("AT command failed, error code: %d", ret);
+		return -EFAULT;
+	}
+
+	return 0;
+}
+
+static void lwm2m_lte_handler_register(void)
+{
+	lte_lc_register_handler(lte_rrc_event_handler);
+}
+
+static void lte_rrc_event_handler(const struct lte_lc_evt *const evt)
+{
+	switch (evt->type) {
+	case LTE_LC_EVT_RRC_UPDATE:
+		if (evt->rrc_mode == LTE_LC_RRC_MODE_CONNECTED) {
+			rrc_connected = true;
+		} else {
+			rrc_connected = false;
+		}
+		LOG_DBG("LTE RRC mode: %s", evt->rrc_mode == LTE_LC_RRC_MODE_IDLE ? "Idle" :
```

Comment:
IMO it'd be cleaner to provide a helper function to convert mode to string, something like https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/ip/net_if.c#L4057.